clean_air/models.py

```diff
@@ -0,0 +1,59 @@
+import dataclasses
+import string
+from dataclasses import dataclass
+from enum import Enum
+from pathlib import Path
+from typing import List
+
+import pyproj
+import shapely.geometry
+
+
+class DataType(Enum):
+    MODEL_GRIDDED = "model_gridded"
+    MODEL_NONGRIDDED = "model_nongridded"
+    OBS_STATIONARY = "obs_stationary"
+    OBS_MOBILE = "obs_mobile"
+    OTHER = "other"
+
+
+@dataclass
+class ContactDetails:
+    title: str = ""
+    first_name: str = ""
+    surname: str = ""
+    email_address: str = ""
+    affiliation: str = ""
+
+
+@dataclass
+class Metadata:
+    dataset_name: str
+    extent: shapely.geometry.Polygon
+    crs: pyproj.CRS = pyproj.CRS("EPSG:4326")
+    description: str = ""
+    data_type: DataType = DataType.OTHER
+    contacts: List[ContactDetails] = dataclasses.field(default_factory=list)
+
+    @property
+    def id(self):
+        return self.dataset_name.lower().replace(string.punctuation, "_")
+
+
+# Assume 1 metadata file for whole dataset, and metadata is consistent across files
+# All files are same filetype
+# All files are netcdf4
+@dataclass
+class DataSet:
+    """"""
+
+    files: List[Path] = None
+    metadata: Metadata = None
+
+    @property
+    def name(self):
+        return self.metadata.dataset_name if self.metadata else ""
+
+    @property
+    def id(self):
+        return self.metadata.id if self.metadata else None
```

Comment:
I don't think I really understand this one; I feel a bit unsure about the metadata class here and others as I feel this is setting values which may be different on input, but I also think lots more is happening elsewhere that I don't understand, so I think we should go over this file when we can.

Reply:
it's basically a convenience method that's equivalent to calling `my_dataset.metadata.id` (but returns `None` if the dataset has no metadata associated with it for some reason).

It seemed a bit odd not to have an `.id` property on dataclass, but I wanted to ensure consistency between this property and the metadata instance